server/migrations/20200223102246-add-deleted-at-index.js

```diff
@@ -1,17 +1,23 @@
-
-
 module.exports = {
   up: async (queryInterface, Sequelize) => {
-    await queryInterface.removeIndex('users', 'users_email');
-    await queryInterface.sequelize.query('ALTER TABLE users DROP CONSTRAINT users_phone_number_key;');
-    await queryInterface.addConstraint('users', ['email', 'deleted_at'], { type: 'UNIQUE' });
-    await queryInterface.addConstraint('users', ['phone_number', 'deleted_at'], { type: 'UNIQUE' });
+    await queryInterface.addConstraint('users', {
+      fields: ['email', 'deleted_at'],
+      type: 'UNIQUE'
+    });
+    await queryInterface.addConstraint('users', {
+      fields: ['phone_number', 'deleted_at'],
+      type: 'UNIQUE'
+    });
   },
 
   down: async (queryInterface, Sequelize) => {
-    await queryInterface.removeIndex('users', ['phone_number', 'deleted_at'], { type: 'UNIQUE' });
-    await queryInterface.removeIndex('users', ['email', 'deleted_at'], { type: 'UNIQUE' });
-    await queryInterface.addIndex('users', 'users_email');
-    await queryInterface.addConstraint('users', 'phone_number');
+    await queryInterface.removeIndex('users',
+      ['phone_number', 'deleted_at'], {
+        type: 'UNIQUE'
+      });
+    await queryInterface.removeIndex('users',
+      ['email', 'deleted_at'], {
+        type: 'UNIQUE'
+      });
```

Comment:
I cannot add a new migration file because I only fixed the existing migration file here because the Sequelize API has changed. The old `await queryInterface.addConstraint('users', ['email', 'deleted_at'], { type: 'UNIQUE' })` does not accept the fields parameter (['email', 'deleted_at']). It was moved to the options parameter. This API Call ist now:
`await queryInterface.addConstraint('users', { fields: ['email', 'deleted_at'],  type: 'UNIQUE' })` Without the Change it was not possible (for me) to migrate up from base(empty db) with the new sequelize version.

See v6 https://sequelize.org/master/class/lib/dialects/abstract/query-interface.js~QueryInterface.html#instance-method-addConstraint 
And v4 https://sequelize.org/v4/class/lib/query-interface.js~QueryInterface.html#instance-method-addConstraint
The parameter is not existing anymore.

Secondly the call to `await queryInterface.removeIndex('users', 'users_email');` and `await queryInterface.sequelize.query('ALTER TABLE users DROP CONSTRAINT users_phone_number_key;');` is useless because this index / constraint is never created in a migration relative to this one before. The equivalent in the down function, why to add an index if it wasn't removed because it was not even there before.